libs/api/domains/user-profile/src/lib/adminUserProfile.resolver.ts

```diff
@@ -0,0 +1,54 @@
+import { Args, Parent, Query, ResolveField, Resolver } from '@nestjs/graphql'
+import { UseGuards } from '@nestjs/common'
+
+import type { User } from '@island.is/auth-nest-tools'
+import {
+  CurrentUser,
+  IdsUserGuard,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import { IdentityClientService } from '@island.is/clients/identity'
+
+import { UserProfileService } from './userProfile.service'
+import { PaginatedUserProfileResponse } from './dto/paginated-user-profile.response'
+import { UserProfile } from './userProfile.model'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Resolver(() => UserProfile)
```

Comment:
The field resolver in this resolver will be added to the main userprofile object, and therefore be listed in the GQL schema for the normal queries. So I think we should create a new `AdminUserProfile` type that extends the default `UserProfile` and add the nudge dates only on the admin type. Then make this resolver resolve the `AdminProfile`.